src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1519,6 +1523,19 @@ private void SetupDirectives(IEnumerable<DirectiveDescriptor> directiveDescripto
             MapDirectives(RemoveTagHelperDirective, SyntaxConstants.CSharp.RemoveTagHelperKeyword);
         }
 
+        private void EnsureDirectiveIsAtStartOfLine()
+        {
+            // 1 is the offset of the @ transition for the directive.
+            if (CurrentStart.CharacterIndex > 1)
+            {
+                var currentDirective = CurrentSymbol.Content;
+                Context.ErrorSink.OnError(
+                    CurrentStart,
+                    Resources.FormatDirectiveMustAppearAtStartOfLine(currentDirective),
```

Comment:
Figured this was new code so the error could live in `Resources.resx`. Other option would be in `LegacyResources.resx`. Not sure how we're choosing to make those decisions.